### PR TITLE
[WIP] feat: max_proposals and block_size limit

### PR DESCRIPTION
**Pull request by zhangsoledad**

### CODE REVIEW

File: core/src/header.rs [hunk 1]
```diff
@@ -90,6 +90,15 @@ impl RawHeader {
     pub fn mut_uncles_count(&mut self) -> &mut u32 {
         &mut self.uncles_count
     }
+
+    // temp
+    pub const fn serialized_size() -> usize {
```
Comment (janx) [inline core/src/header.rs]: Let's improve this to remove hard-coded calculations.

Comment (janx) [reply core/src/header.rs]: I suggest the name `normalized_size`, as this size is used for block size limit calculation, not a size of real serialization.

Comment (janx) [reply core/src/header.rs]: Would it be better to abstract the logic into `NormalizedSizeCalculator` to calculate `normalized_size` for different structs?

Comment (doitian) [reply core/src/header.rs]: We'll use CFB later, let's use the temp quick solution first to verify our assumption.

**Comment by doitian** [comment]
I'll continue the work

**Pull request closed without merging**